build.sbt

```diff
@@ -1,49 +1,109 @@
-      <id>cb372</id>
-      <name>Chris Birchall</name>
-      <url>https://github.com/cb372</url>
-    </developer>
-  </developers>
-)
-pomIncludeRepository := { _ => false }
+val commonSettings = Seq(
+  scalaVersion := "2.11.8",
+  crossPaths := false,
+  releasePublishArtifactsAction := PgpKeys.publishSigned.value,
+  organization := "com.gu",
+  licenses := Seq("Apache v2" -> url("http://www.apache.org/licenses/LICENSE-2.0.html"))
+) ++ mavenSettings
+
+/**
+  * Root project
+  */
+lazy val root = Project(id = "root", base = file("."))
+  .settings(commonSettings)
+  .aggregate(models, json)
+  .settings(
+    publishArtifact := false,
+    releaseProcess := Seq(
+      checkSnapshotDependencies,
+      inquireVersions,
+      runClean,
+      runTest,
+      setReleaseVersion,
+      commitReleaseVersion,
+      tagRelease,
+      publishArtifacts,
+      setNextVersion,
+      commitNextVersion,
+      releaseStepCommand("sonatypeReleaseAll"),
+      pushChanges
+    )
+  )
+
+/**
+  * Thrift models project
+  */
+lazy val models = Project(id = "content-api-models", base = file("models"))
+  .settings(commonSettings)
+  .disablePlugins(ScroogeSBT)
+  .settings(
+    description := "Scala models for the Guardian's Content API",
+    unmanagedResourceDirectories in Compile += { baseDirectory.value / "src/main/thrift" },
+    libraryDependencies ++= Seq(
+      "org.apache.thrift" % "libthrift" % "0.9.2",
+      "com.twitter" %% "scrooge-core" % "3.20.0",
+      "com.gu" % "story-packages-model-thrift" % "1.0.3",
+      "com.gu" % "content-atom-model-thrift" % "1.0.0"
+    )
+  )
+
+/**
+  * JSON parser project
+  */
+lazy val json = Project(id = "content-api-models-json", base = file("json"))
+  .dependsOn(models)
+  .settings(commonSettings)
+  .settings(
+    description := "Json parser for the Guardian's Content API models",
+    javacOptions ++= Seq("-source", "1.7", "-target", "1.7"),
+    scalacOptions ++= Seq("-deprecation", "-unchecked"),
+
+    scroogeThriftOutputFolder in Compile := sourceManaged.value / "thrift",
+    scroogeThriftSourceFolder in Compile := baseDirectory.value / "../models/src/main/thrift",
+    scroogeThriftDependencies in Compile ++= Seq(
+      "content-api-models",
+      "story-packages-model-thrift",
+      "content-atom-model-thrift"
+    ),
+    // See: https://github.com/twitter/scrooge/issues/199
+    scroogeThriftSources in Compile ++= {
+      (scroogeUnpackDeps in Compile).value.flatMap { dir => (dir ** "*.thrift").get }
+    },
+
+    libraryDependencies ++= Seq(
+      "org.json4s" %% "json4s-native" % "3.3.0",
+      "org.json4s" %% "json4s-ext" % "3.3.0",
+      "joda-time" % "joda-time" % "2.3",
+      "net.databinder.dispatch" %% "dispatch-core" % "0.11.3",
+      "org.scalatest" %% "scalatest" % "2.2.1" % "test",
+      "com.google.guava" % "guava" % "19.0" % "test"
+    ),
+
+    mappings in (Compile, packageBin) ~= { _.filter( f => f._1.getAbsolutePath.contains("com/gu/contentapi/json"))}
```

Comment:
This should be done for `packageSrc` as well, so we don't include the unnecessary files in the sources jar.

```
val jsonPackageFilter = (fp: (java.io.File, String)) => fp._1.getAbsolutePath.contains("com/gu/contentapi/json")

...

mappings in (Compile, packageBin) ~= { _.filter(jsonPackageFilter) },
mappings in (Compile, packageSrc) ~= { _.filter(jsonPackageFilter) }
```

For the javadoc jar, it's more complicated, as we can't just exclude files. The HTML in the javadoc will refer to files that we have excluded. 

I think we should just not publish a javadoc jar. It's not mandatory for Maven Central anyway.

```
publishArtifact in (Compile, packageDoc) := false
```